service/controller/clusterapi/v29/resource/ipam/create.go

```diff
@@ -85,49 +83,65 @@ func (r *Resource) getReservedNetworks(ctx context.Context) ([]net.IPNet, error)
 	g := &errgroup.Group{}
 
 	g.Go(func() error {
-		r.logger.LogCtx(ctx, "level", "debug", "message", "finding allocated subnets from VPCs")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "finding allocated subnets from AWSConfig CRs")
 
-		subnets, err := getVPCSubnets(ctx)
+		subnets, err := r.getSubnetsFromAWSConfigs(ctx)
 		if err != nil {
 			return microerror.Mask(err)
 		}
 		mutex.Lock()
 		reservedSubnets = append(reservedSubnets, subnets...)
 		mutex.Unlock()
 
-		r.logger.LogCtx(ctx, "level", "debug", "message", "found allocated subnets from VPCs")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "found allocated subnets from AWSConfig CRs")
 
 		return nil
 	})
 
 	g.Go(func() error {
-		r.logger.LogCtx(ctx, "level", "debug", "message", "finding allocated subnets from AWSConfig CRs")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "finding allocated subnets from Cluster CRs")
 
-		subnets, err := getAWSConfigSubnets(r.g8sClient)
+		subnets, err := r.getSubnetsFromClusters(ctx)
 		if err != nil {
 			return microerror.Mask(err)
 		}
 		mutex.Lock()
 		reservedSubnets = append(reservedSubnets, subnets...)
 		mutex.Unlock()
 
-		r.logger.LogCtx(ctx, "level", "debug", "message", "found allocated subnets from AWSConfig CRs")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "found allocated subnets from Cluster CRs")
 
 		return nil
 	})
 
 	g.Go(func() error {
-		r.logger.LogCtx(ctx, "level", "debug", "message", "finding allocated subnets from Cluster CRs")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "finding allocated subnets from MachineDeployment CRs")
 
-		subnets, err := getClusterSubnets(r.cmaClient)
+		subnets, err := r.getSubnetsFromMachineDeployments(ctx)
```

Comment:
This one got actually added. We want to also collect subnets from machine deployments.